neuralcompression/layers/_generalized_divisive_normalization.py

```diff
@@ -0,0 +1,231 @@
+                    self._reparameterized_alpha.initial_value,
+                )
+        else:
+            if isinstance(alpha_parameter, Parameter):
+                self.alpha = alpha_parameter
+            else:
+                alpha_parameter = torch.tensor(alpha_parameter)
+
+                self.alpha = Parameter(alpha_parameter)
+
+        if beta_parameter is None:
+            if beta_initializer is None:
+                beta_initializer = functools.partial(
+                    lambda x: torch.ones(x),
+                )
+
+            self._reparameterized_beta = NonNegativeParameterization(
+                beta_initializer(channels),
+                minimum=1e-6,
+            )
+
+            if self._reparameterized_beta.initial_value is not None:
+                self.beta = Parameter(
+                    self._reparameterized_beta.initial_value,
+                )
+        else:
+            if isinstance(beta_parameter, Parameter):
+                self.beta = beta_parameter
+            else:
+                beta_parameter = torch.tensor(beta_parameter)
+
+                self.beta = Parameter(beta_parameter)
+
+        if epsilon_parameter is None:
+            if epsilon_initializer is None:
+                epsilon_initializer = functools.partial(
+                    lambda x: torch.ones(x),
+                )
+
+            self._reparameterized_epsilon = NonNegativeParameterization(
+                epsilon_initializer(channels),
+                minimum=1e-6,
+            )
+
+            if self._reparameterized_epsilon.initial_value is not None:
+                self.epsilon = Parameter(
+                    self._reparameterized_epsilon.initial_value,
+                )
+        else:
+            if isinstance(epsilon_parameter, Parameter):
+                self.epsilon = epsilon_parameter
+            else:
+                epsilon_parameter = torch.tensor(epsilon_parameter)
+
+                self.epsilon = Parameter(epsilon_parameter)
+
+        if gamma_parameter is None:
+            if gamma_initializer is None:
+                gamma_initializer = functools.partial(
+                    lambda x: 0.1 * torch.eye(x),
+                )
+
+            self._reparameterized_gamma = NonNegativeParameterization(
+                gamma_initializer(channels),
+                minimum=0,
+            )
+
+            if self._reparameterized_gamma.initial_value is not None:
+                self.gamma = Parameter(
+                    self._reparameterized_gamma.initial_value,
+                )
+        else:
+            if isinstance(gamma_parameter, Parameter):
+                self.gamma = gamma_parameter
+            else:
+                gamma_parameter = torch.tensor(gamma_parameter)
+
+                self.gamma = Parameter(gamma_parameter)
+
+    def forward(self, x: Tensor) -> Tensor:
+        _, channels, _, _ = x.size()
+
+        if self._rectify:
+            x = torch.nn.functional.relu(x)
```

Comment:
We don't need this here - we should just make users use their own `torch.nn.ReLU()` layer.

Reply:
Good point.

Reply:
Removed!